### PR TITLE
Add type hints

### LINKED ISSUE #103: Add type hints
Similar to apispec, can you please consider to add typehints?
For that a new release of apispec releasing the type hints would be great!

### PR DESCRIPTION
**Pull request by kasium**
Please note that a upgrade of black was needed due https://github.com/psf/black/issues/2964

Closes #103

### CODE REVIEW

**Comment by kasium** [comment]
@lafrech after fighting with the tests for some time I think it's the best if load_operations_from_docstring would accept a None for a docstring. It is capable to handle it and an assert would break existing applications

**Comment by kasium** [comment]
@lafrech I just picking up on long running PRs. Are you still willing to consider this feature? I would then update the code and try to make the tests green

**Comment by lafrech** [comment]
Sorry about the lack of feedback. Thanks for the ping.

Yes, this contribution is welcome.

Thanks.

-- 
Jérôme

**Comment by kasium** [comment]
@lafrech there is an error in Azure
```
##[warning]An image label with the label ubuntu-18.04 does not exist.
,##[error]The remote provider was unable to process the request.
```

**Comment by lafrech** [comment]
Yep. Let's go on with #108 first.

**Comment by lafrech** [comment]
You may rebase this now that CI is fixed.

Thanks !

**Comment by kasium** [comment]
@lafrech sorry for the radio silence from my side. CI is now green 😄

File: src/apispec_webframeworks/bottle.py [hunk 1]
```diff
@@ -37,11 +38,11 @@ class BottlePlugin(BasePlugin):
     """APISpec plugin for Bottle"""
 
     @staticmethod
-    def bottle_path_to_openapi(path):
+    def bottle_path_to_openapi(path: str) -> str:
         return RE_URL.sub(r"{\1}", path)
 
     @staticmethod
-    def _route_for_view(app, view):
+    def _route_for_view(app: Bottle, view):
```
Comment (lafrech) [inline src/apispec_webframeworks/bottle.py:45]: Missing return type

**Review by lafrech — APPROVED** [review]
Thanks. LGTM.

I spotted a missing return value and you could comment on why the "ignore type" lines, but it's already much better that nothing so I'm willing to merge this.

**Comment by kasium** [comment]
I removed all type ignores beside one where I added a comment explaining it.
To simplify the code, I removed support for the 2017 released tornado 4.4

**Comment by lafrech** [comment]
Thanks!

**Pull request merged**